### PR TITLE
Gives chitinous armor durability

### PR DESCRIPTION
**Pull request by Coolrune206**
## What Does This PR Do

Replaces the old removal method of chitinous armor (using a saw on a prone target over 15 seconds) with an armor durability system. It functions as follows:
Every time the armor is deployed, it sets its durability to 600. Damage done to the Changeling is also done to the armor, with modifications dependent on damage type and certain weapons.
Charged Plasma Pistol shots do 200 flat damage to the armor. The Plasma Pistol is designed to be a niche weapon that destroys shields and armor, so giving it some targeted functionality here seems like a good idea. 
Plasma Cutters do 4x damage to armor, so regular does 20, advanced does 28, and the mech one does 40.
Circular saws do 25 flat damage to the armor. This is because previously, circular saws were the only tools that could remove the armor apart from the cling doing so - so having them be slightly more effective makes sense.
Ranged/melee burn attacks (welders, lasers, etc) will do double damage to armor durability.
All other ranged/melee attacks (mostly brute though this could technically include wt toxin bullets and the like) will have an unmodified damage value being applied to the armor.
Once the armor's durability falls below zero, it falls off the changeling in shreds. 
The armor durability value is not final and can be increased if requested by the balance team, these numbers are just what I floated by Hal earlier.
## Why It's Good For The Game

There's quite a few reasons for this change, so I'll go through them in order:

There's already a currently existing interaction for removing the suit with a saw but it's not very practical (requires the changeling to sit still for 15 seconds, so it'd have to be dead and it already revives in 40 seconds so good luck getting the armor off and doing anything else meaningful before it stands up and re-equips it), and the interaction is not very known. 

The armor itself is quite powerful, being an ability that costs chemicals once to deploy and barely slowing down chemical regen while it's active, with good armor values to boot. Having it degrade over combat is one way to actually have it have implications to the Changeling's chemical economy, as you'll have to choose whether to re-apply it when it falls off or use another ability, instead of just having it up forever once you've clicked it once.

Lastly, this armor actually makes it quite hard to properly destroy the Changeling once it's been activated. Now that decapitation isn't an option versus Clings, all a Changeling has to do is break the cremator and pop this armor and you suddenly have very few options of actually dealing with them, as this armor pretty much prevents you from putting them in a gibber as the armor obstructs most of your items and you can't gibber people wearing said items. Your only options at that point are repairing the cremator which takes a bit and will probably get broken again, shuttle-gibbing them which (while it's very cool) is generally not practical most of the time, or carrying around SR pills in your pockets to kill Changelings who have made it very hard to otherwise kill them. I don't think the latter practice is at all healthy as it basically removes the whole point of making Changelings harder to kill elsewise, so I think making it more practical to kill them through means like the gibber (which still leaves the Cling a chance to get up and escape before they get shoved in) is better.
## Images of changes

https://user-images.githubusercontent.com/71326864/217941509-16e26c18-8ea9-48f4-949e-be02d90fd920.mp4
## Testing

Tested the cling armor with plasma pistol, circular saw, batong, WT autorifle, laser gun, welder, all applied damage as it should, armor broke when armordurability hit zero. Visible in the above video.
## Changelog
tweak: Changeling chitinous armor now has durability, and can break once damaged enough.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 1]
```diff
@@ -479,61 +480,32 @@
...
+	else
+		var/obj/item/I = hitby
+		if(istype(I, /obj/item/circular_saw))
+			armordurability -= 25 // saws used to cut away armor through an interaction, so let's keep them somewhat more effective
+		else if(I.damtype == BURN)
+			armordurability -= damage * 2
+		else
+			armordurability -= damage
+	if(armordurability <= 0)
+		visible_message("<span class='warning'>[owner]'s chitinous armor collapses in clumps onto the ground.</span>",)
```
Comment (Henri215) [inline code/modules/antagonists/changeling/powers/mutations.dm:502]:
```suggestion
		visible_message("<span class='warning'>[owner]'s chitinous armor collapses in clumps onto the ground.</span>")
```

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 2]
```diff
@@ -479,61 +480,32 @@
...
+		armor_durability -= 200 // the plasma pistol is designed to be a niche tool for blasting through shields, may as well let it cut away armor
+	else if(istype(hitby, /obj/item/projectile))
+		var/obj/item/projectile/P = hitby
+		if(P.damage_type == BURN)
+			armor_durability -= damage * 2
+		else
+			armor_durability -= damage
+	else
+		var/obj/item/I = hitby
+		if(istype(I, /obj/item/circular_saw))
```
Comment (lewcc) [inline code/modules/antagonists/changeling/powers/mutations.dm:495]: For the original saw removal PR, it was pretty highly requested that chainsaws be good at removing the armor. Think you could keep that in?

Comment (Coolrune206) [reply code/modules/antagonists/changeling/powers/mutations.dm:495]: Due to their melee damage they should be rather good at removing the armor anyways.

**Comment by matttheficus** [comment]
As much as the funny "plasma cutter bad" thing for Security, thoughts on it also dealing a bit bonus damage to the armor? A la Necromorphs from Dead Space.

The fact it prevents gibber use is very annoying. I like this change as a whole.

**Comment by Coolrune206** [comment]
Plasma Cutters now do 4x damage to the armor. This sounds like a lot, but their base damage is 5 for the regular, 7 for the advanced, and 10 for the exosuit mounted one.

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 3]
```diff
@@ -479,61 +480,34 @@
...
+		else
+			armor_durability -= damage
+	if(armor_durability <= 0)
+		visible_message("<span class='warning'>[owner]'s chitinous armor collapses in clumps onto the ground.</span>")
+		new /obj/effect/decal/cleanable/shreds(owner.loc)
+		// just unequip them since they qdel on drop
+		owner.unEquip(src, TRUE, TRUE)
+		if(istype(owner.head, /obj/item/clothing/head/helmet/changeling))
+			owner.unEquip(owner.head, TRUE, TRUE)
+	return
```
Comment (S34NW) [inline code/modules/antagonists/changeling/powers/mutations.dm:510]: You don't need this return, but it's a minor style thing

**Review by S34NW — APPROVED** [review]

**Comment by Peakwastaken** [comment]
How about giving it a scaling with the amount of genomes gathered ? I really like this mechanic from the shield as it gives a more unga ability some more time and work required to get the most out of it, with some chances to RP while gathering genomes, as well as more meaning to the genome gathering part of clings

How about base armor of 200+genome*20 with a cap at 30 genomes for 400 armor at 10, 600 at 20, 800 at 30 ?

Do agree with this fully though, i'm thinking 600 might be a tiiiinyy too little so that's why I'm proposing a possibility for more with some work behind it

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 4]
```diff
@@ -479,61 +480,34 @@
...
-	if(istype(owner.head, /obj/item/clothing/head/helmet/changeling))
-		owner.unEquip(owner.head, TRUE, TRUE)
-
-	return TRUE
-
+	if(istype(hitby, /obj/item/projectile/energy/charged_plasma))
+		armor_durability -= 200 // the plasma pistol is designed to be a niche tool for blasting through shields, may as well let it cut away armor
+	else if(istype(hitby, /obj/item/projectile/plasma))
+		armor_durability -= damage * 4
+	else if(istype(hitby, /obj/item/projectile))
```
Comment (SteelSlayer) [inline code/modules/antagonists/changeling/powers/mutations.dm:489]: Put these in `bullet_act()`

**Review by SteelSlayer — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/changeling/powers/mutations.dm (hunk 4 above)
Comment (Coolrune206) [reply code/modules/antagonists/changeling/powers/mutations.dm:489]: Can do, but for my own curiosity what's the difference?

Comment (SteelSlayer) [reply code/modules/antagonists/changeling/powers/mutations.dm:489]: Nevermind.

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**